static/app/views/profiling/continuousProfileProvider.tsx

```diff
@@ -0,0 +1,142 @@
+
+interface ContinuousProfileQueryParams {
+  end: string;
+  profiler_id: string;
+  start: string;
+}
+
+function fetchContinuousProfileFlamegraph(
+  api: Client,
+  query: ContinuousProfileQueryParams,
+  projectSlug: Project['slug'],
+  orgSlug: Organization['slug']
+): Promise<Profiling.ProfileInput> {
+  return api
+    .requestPromise(`/organizations/${orgSlug}/profiling/chunks/`, {
+      method: 'GET',
+      query: {
+        ...query,
+        project: projectSlug,
+      },
+      includeAllArgs: true,
+    })
+    .then(([data]) => data);
+}
+
+type ContinuousProfileProviderValue = RequestState<Profiling.ProfileInput>;
+export const ContinuousProfileContext =
+  createContext<ContinuousProfileProviderValue | null>(null);
+
+export function useContinuousProfile() {
+  const context = useContext(ContinuousProfileContext);
+  if (!context) {
+    throw new Error('useContinuousProfile was called outside of ProfileProvider');
+  }
+  return context;
+}
+
+function isValidDate(date: string): boolean {
+  return !isNaN(Date.parse(date));
+}
+
+function getContinuousChunkQueryParams(
+  query: string
+): ContinuousProfileQueryParams | null {
+  const qs = new URLSearchParams(query);
+  const start = qs.get('start');
+  const end = qs.get('end');
+  const profiler_id = qs.get('profilerId');
+
+  if (!start || !end || !profiler_id) {
+    return null;
+  }
+
+  if (!isValidDate(start) || !isValidDate(end)) {
+    return null;
+  }
+
+  return {
+    start,
+    end,
+    profiler_id,
+  };
+}
+
+interface ContinuousFlamegraphViewProps {
+  children: React.ReactNode;
+}
+
+function ContinuousProfileProvider(
+  props: ContinuousFlamegraphViewProps
+): React.ReactElement {
+  const api = useApi();
+  const params = useParams();
+  const organization = useOrganization();
+  const projects = useProjects();
+
+  const [profiles, setProfiles] = useState<RequestState<Profiling.ProfileInput>>({
+    type: 'initial',
+  });
+
+  useLayoutEffect(() => {
+    if (!params.projectId) {
+      return undefined;
+    }
+
+    const chunkParams = getContinuousChunkQueryParams(window.location.search);
+    const project = projects.projects.find(p => p.slug === params.projectId);
+
+    if (!chunkParams) {
+      Sentry.captureMessage(
+        'Failed to fetch continuous profile - invalid query parameters.'
+      );
+      return undefined;
+    }
+    if (!project) {
+      Sentry.captureMessage('Failed to fetch continuous profile - project not found.');
+      return undefined;
+    }
+
+    setProfiles({type: 'loading'});
+
+    fetchContinuousProfileFlamegraph(api, chunkParams, project.id, organization.slug)
+      .then(p => {
+        setProfiles({type: 'resolved', data: p});
+      })
+      .catch(err => {
+        // XXX: our API client mock implementation does not mimick the real
+        // implementation, so we need to check for an empty object here. #sad
+        const isEmptyObject = err.toString() === '[object Object]';
```

Comment:
Is this used in tests somehow? The only test that errors I see doesn't reach this point.

Reply:
It is in the old test. Let me remove it here, we dont need it rn